### DISCUSSION TITLE
pgxpool & GetSSLPassword

### DISCUSSION THREAD

**Discussion by jwc-clinnection** [jackc/pgx#1823] [Q&A]
I've looked all over and can't seem to find how to use pgxpool & GetSSLPassword. Would someone please help me? Thank you.

**Comment by jackc** [comment]
I suppose we could add a pgxpool.ParseConfigWithOptions, but at least for now I think you can use pgxpool.ParseConfig and then overwrite the ConnConfig field of the pgxpool.Config with pgx.ParseConfigWithOptions.

**Comment by jwc-clinnection** [comment]
Is this the correct way of using GetSSLPassword with pgxpool? Thank you in advance. 

```
func getSslPassphrase(ctx context.Context) string {
	return "ssl_passphrase"
}

func main() {
	url := "postgres://username:password@localhost:5432/database_name"

	parseConfigOptions := pgconn.ParseConfigOptions{
		GetSSLPassword: getSslPassphrase,
	}

	configWithOptions, _ := pgconn.ParseConfigWithOptions(url, parseConfigOptions)

	config := pgxpool.Config{
		ConnConfig: &pgx.ConnConfig{
			Config: *configWithOptions,
		},
	}

	conn, err := pgxpool.NewWithConfig(context.Background(), &config)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Unable to connect to database: %v\n", err)
		os.Exit(1)
	}
}
```

**Reply by jackc** [reply]
Close, but the pgxpool.Config and pgx.ConnConfig need to be created with a Parse* function. This is to ensure that default fields are set properly.

I would suggest getting it to work with a single pgx.Conn before getting attempting to get it to work with a pgxpool.Pool.
